server/src/test/java/org/elasticsearch/search/aggregations/metrics/ScriptedMetricAggregatorTests.java

```diff
@@ -181,6 +200,46 @@ public static void initMockScripts() {
            state.put("selfRef", state);
            return state;
         });
+        SCRIPTS.put("initScriptMakingArray", params -> {
+            Map<String, Object> state = (Map<String, Object>) params.get("state");
+            state.put("array", new String[] {"foo", "bar"});
+            state.put("collector", new ArrayList<Integer>());
+            return state;
+         });
+    }
+
+    private CircuitBreakerService circuitBreakerService;
+
+    @Before
+    public void mockBreaker() {
```

Comment:
We could check this for all aggs in the `AggregatorTestCase` directly ?

Reply:
Would you be ok if I moved it in there in a followup? I think there are quite a few fiddly things involved in getting that right.

Reply:
Sure, it might uncovered some other bugs so be prepared ;).